src/ogr/feature.jl

```diff
@@ -351,49 +376,55 @@ end
 #           pfSecond,pnTZFlag)
 # end
 
-getdefault(feature::Feature, i::Integer) = getdefault(getfielddefn(feature, i))
-
-const _FETCHFIELD = Dict{GDAL.OGRFieldType, Function}(
-    GDAL.OFTInteger         => asint,           #0
-    GDAL.OFTIntegerList     => asintlist,       #1
-    GDAL.OFTReal            => asdouble,        #2
-    GDAL.OFTRealList        => asdoublelist,    #3
-    GDAL.OFTString          => asstring,        #4
-    GDAL.OFTStringList      => asstringlist,    #5
- # const OFTWideString =                (UInt32)(6)
- # const OFTWideStringList =            (UInt32)(7)
-    GDAL.OFTBinary          => asbinary,        #8
- # const OFTDate =                      (UInt32)(9)
- # const OFTTime =                      (UInt32)(10)
-    GDAL.OFTDateTime        => asdatetime,      #11
-    GDAL.OFTInteger64       => asint64,         #12
-    GDAL.OFTInteger64List   => asint64list      #13
- # const OFTMaxType =                   (UInt32)(13)
- )
+function getdefault(feature::Feature, i::Integer)::String
+    return getdefault(getfielddefn(feature, i))
+end
 
 function getfield(feature::Feature, i::Integer)
-    if isfieldset(feature, i)
+    _FETCHFIELD = Dict{OGRFieldType,Function}(
+        OFTInteger => asint,           #0
+        OFTIntegerList => asintlist,       #1
+        OFTReal => asdouble,        #2
+        OFTRealList => asdoublelist,    #3
+        OFTString => asstring,        #4
+        OFTStringList => asstringlist,    #5
+        # const OFTWideString =                (UInt32)(6)
+        # const OFTWideStringList =            (UInt32)(7)
+        OFTBinary => asbinary,        #8
+        OFTDate => asdatetime,      #9
+        OFTTime => asdatetime,      #10
+        OFTDateTime => asdatetime,      #11
+        OFTInteger64 => asint64,         #12
+        OFTInteger64List => asint64list,     #13
+    )
+    return if isfieldset(feature, i)
         _fieldtype = gettype(getfielddefn(feature, i))
         _fetchfield = get(_FETCHFIELD, _fieldtype, getdefault)
-        return _fetchfield(feature, i)
+        _fetchfield(feature, i)
     else
-        return getdefault(feature, i)
+        getdefault(feature, i)
     end
 end
 
-getfield(feature::Feature, name::Union{AbstractString, Symbol}) =
-    getfield(feature, findfieldindex(feature, name))
+function getfield(feature::Feature, name::Union{AbstractString,Symbol})
+    i = findfieldindex(feature, name)
+    return if i == -1
+        nothing
+    else
+        getfield(feature, i)
+    end
+end
```

Comment:
@evetion (and others), just wanted to call out this snippet, and get your thoughts on returning `nothing` (versus e.g. `missing`) when `i == -1`. (See #177 for context.)

Reply:
I think `missing` would be a better choice here. In the julia docs there is [a FAQ entry](https://docs.julialang.org/en/v1/manual/faq/#Nothingness-and-missing-values-1), and it states:

> To represent missing data in the statistical sense (NA in R or NULL in SQL), use the missing object.

The NULL in the example of #177 seems to represent missing data to me. Especially if we see Features as rows in a Table / records in a dataset.

`findfieldindex`, if it was written in Julia, should probably return `nothing` instead of `-1`, like `findfirst`. But the index truly doesn't exist, whereas the data can be viewed as missing (the field name exists after all).

Reply:
Switched to `missing` in https://github.com/yeesian/ArchGDAL.jl/pull/179/commits/83b70259519581b1ca8070284f5c56d2f3d9e053, and opened https://github.com/yeesian/ArchGDAL.jl/issues/192 to ensure that the codebase is consistent in semantics after this PR.

Reply:
Leaving this comment thread unresolved for you to go through it. Feel free to let me know if I might have missed anything.